### PR TITLE
Add docs/release/0.14.0.rst

### PR DESCRIPTION
**Pull request by paugier**
Needed for #385

### CODE REVIEW

**Comment by hgomersall** [comment]
Thanks for this.

Did I contribute? I can't remember what!

**Comment by paugier** [comment]
I think you did :-)  https://github.com/pyFFTW/pyFFTW/pull/354

**Review by hgomersall — DISMISSED** [review]

File: docs/release/0.14.0.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+============================
```
Comment (hgomersall) [inline docs/release/0.14.0.rst:1]: This top line seems to be produced as-is in git-flavoured markdown.

File: docs/release/0.14.0.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+pyFFTW v0.14.0 release notes
+============================
```
Comment (hgomersall) [inline docs/release/0.14.0.rst:3]: Do we want a top level summary of the main changes, or are we happy with the pull requests list describing those?

File: docs/release/0.14.0.rst (hunk 1 above)
Comment (paugier) [reply docs/release/0.14.0.rst]: I removed this line.

File: docs/release/0.14.0.rst (hunk 2 above)
Comment (paugier) [reply docs/release/0.14.0.rst:2]: There is already a short paragraph on the main changes, in particular "A highlight of this release is compatibility with Cython 3, Python 3.12 and Numpy 2.0.". IMHO, it's enough. Do you want to add more?

**Comment by paugier** [comment]
I think this PR should be merged because there are other things to do and we should work on a new version targeting Python>=3.10 and supporting 3.13 (to follow Numpy and co).

**Review by hgomersall — APPROVED** [review]

**Comment by hgomersall** [comment]
Changes approved - merge when ready!

**Pull request merged**